src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -321,7 +311,7 @@ private ChatCompletionsOptions ToAzureAIOptions(IEnumerable<ChatMessage> chatCon
                         default:
                             if (prop.Value is not null)
                             {
-                                byte[] data = JsonSerializer.SerializeToUtf8Bytes(prop.Value, ToolCallJsonSerializerOptions.GetTypeInfo(typeof(object)));
+                                byte[] data = JsonSerializer.SerializeToUtf8Bytes(prop.Value, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(object)));
```

Comment:
I'd forgotten about `AdditionalProperties`. Hardcoding to the default options effectively forces known types only in AOT.

Reply:
I suspect primitives is sufficient. We can always add an overload later that takes a JSO if it proves important. Or you think it should be done now?

Reply:
I suspect there always will be a need for storing something less trivial than just primitives.

Reply:
Quite possibly, but this in particular is about properties that should be sent up to the service as extra properties that a given service understands:
https://learn.microsoft.com/en-us/rest/api/aifoundry/modelinference/#extensibility
and those are generally going to be primitives, like float, int, or string.

(Also, the Azure.AI.Inference library itself is not yet AOT friendly.)

Reply:
What about `float[]`, `int?`, `int?[]`, or `string[]`? If we're drawing a line in the sand we probably need to document it (and quite possibly reject values whose type is not supported at runtime). Otherwise we find ourselves in a situation where clients that work without issue in CoreCLR end up failing in Native AOT without a workaround.

Reply:
? What about float[], int?, int?[], or string[]?

`int?` boxes the same as `int`, but regardless it's already represented in the default options.

If we think `float[]`, `int?[]`, or `string[]` will be desirable, we can just add them to AIJsonUtilities.